src/main/java/com/example/repository/BookRepositoryImpl.java

```
@@ -0,0 +1,43 @@
+package com.example.repository;
+
+import com.example.model.Book;
+import jakarta.persistence.EntityManager;
+import jakarta.persistence.EntityManagerFactory;
+import jakarta.persistence.EntityTransaction;
+import java.util.List;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Repository;
+
+@Repository
+public class BookRepositoryImpl implements BookRepository {
+    private final EntityManagerFactory factory;
+
+    @Autowired
+    public BookRepositoryImpl(EntityManagerFactory factory) {
+        this.factory = factory;
+    }
+
+    @Override
+    public Book save(Book book) {
+        EntityTransaction transaction = null;
+        try (EntityManager entityManager = factory.createEntityManager()) {
+            transaction = entityManager.getTransaction();
+            transaction.begin();
+            entityManager.persist(book);
+            transaction.commit();
+            return book;
+        } catch (RuntimeException e) {
+            if (transaction != null && transaction.isActive()) {
+                transaction.rollback();
+            }
+            throw e;
```

let's throw an error with clear message here